File: DAL/Dtos/AddressDto.cs
namespace DAL.Dtos
{
    public class AddressDto
    {
        public int Id { get; set; }

        public ApartmentDto? Apartment { get; set; }

        public StreetDto? Street { get; set; }

        public string? ApartmentNumber { get; set; }

        public string? HouseNumber { get; set; }

        public int BuildingNumber { get; set; }

        public int EntranceNumber { get; set; }
    }
}

File: Utilities/ClientApp/src/app/core/Models/AddressDto.ts
import { ApartmentDto } from "./ApartmentDto";
import { StreetDto } from "./StreetDto";

export interface AddressDto {
    id?: number;
    street: StreetDto;
    apartment?: ApartmentDto;
    apartmentNumber: string;
    houseNumber: string;
    buildingNumber: number;
    entranceNumber: number;
}